src/alchemlyb/parsing/lammps.py

```diff
@@ -0,0 +1,1303 @@
+    elif units == "nano":
+        # E in attogram-nanometer^2/nanosecond^2, vol in nm^3, pressure in attogram/(nanometer-nanosecond^2)
+        scaling_factor = 1
+    else:
+        raise ValueError(
+            "LAMMPS unit type, {}, is not supported. Supported types are: cgs, electron,"
+            " lj, metal, micro, nano, real, si".format(units)
+        )
+
+    return scaling_factor
+
+
+def _tuple_from_filename(filename, separator="_", indices=[2, 3], prec=4):
+    r"""Pull a tuple representing the lambda values used, as defined by the filenames.
+
+    Parameters
+    ----------
+    filename : str
+        Filename and path
+    separator : str, default="_"
+        Separator used to breakup the filename. The choice in ``indices`` is dependent on this choice.
+    indices : list, default=[2, 3]
+        Indices used to pull :math:`\lambda` and :math:`\lambda'`
+    prec : int, default=4
+        Number of decimal points in the output.
+
+    Returns
+    -------
+    tuple[float]
+        Tuple of lambda values
+
+    .. versionadded:: 2.4.1
+
+    """
+    filename = filename.replace(".bz2", "").replace(".gz", "")
+    name_array = ".".join(os.path.split(filename)[-1].split(".")[:-1]).split(separator)
+    try:
+        value1 = float(name_array[indices[0]])
+    except ValueError:
+        raise ValueError(
+            f"Entry, {indices[0]} in filename cannot be converted to float: {name_array[indices[0]]}"
+        )
+
+    try:
+        value2 = float(name_array[indices[1]])
+    except ValueError:
+        raise ValueError(
+            f"Entry, {indices[1]} in filename cannot be converted to float: {name_array[indices[1]]}"
+        )
+
+    return (
+        round(value1, prec),
+        round(value2, prec),
+    )
+
+
+def _lambda_from_filename(filename, separator="_", index=-1, prec=4):
+    r"""Pull the :math:`\lambda'` value, as defined by the filenames.
+
+    Here :math:`\lambda'` is the scaling value applied to a configuration that is equilibrated to
+    a different value of :math:`\lambda`.
+
+    Parameters
+    ----------
+    filename : str
+        Filename and path
+    separator : str, default="_"
+        Separator used to breakup the filename. The choice in ``index`` is dependent on this choice.
+    index : list, default=1
+        Index used to pull :math:`\lambda'`
+    prec : int, default=4
+        Number of decimal points in the output.
+
+    Returns
+    -------
+    float
+        Lambda prime value
+
+    .. versionadded:: 2.4.1
+
+    """
+    filename = filename.replace(".bz2", "").replace(".gz", "")
+    name_array = ".".join(os.path.split(filename)[-1].split(".")[:-1]).split(separator)
+    try:
+        value = float(name_array[index])
+    except:
+        raise ValueError(
+            f"Entry, {index} in filename cannot be converted to float: {name_array[index]}"
+        )
+    return round(value, prec)
+
+
+def _get_bar_lambdas(fep_files, indices=[2, 3], prec=4, force=False):
```

Comment:
Haven't checked the coverage yet but this is the part where I expect high code coverage